x-pack/plugins/infra/public/components/logging/log_minimap/time_ruler.tsx

```diff
@@ -19,11 +20,33 @@ interface TimeRulerProps {
   width: number;
 }
 
+const ONE_MINUTE = 60000;
+
+/**
+ * In order to get the correct timestamp for a custom timezone
+ * we need, from the current timestamp, to add:
+ * 1. The current timezone offset in milliseconds (positive or negative)
+ * 2. The UTC offset in milliseconds (positive or negative)
+ * This guarantees the timestamp we return is related to the timezone set globally in moment-js.
+ */
+const applyTimezoneOffset = (utcTimestamp: number) => {
+  const timezoneOffsetMillis = new Date(utcTimestamp).getTimezoneOffset() * ONE_MINUTE;
+  const utcOffsetMillis = moment(utcTimestamp).utcOffset() * ONE_MINUTE;
+
+  return utcTimestamp + timezoneOffsetMillis + utcOffsetMillis;
+};
+
 export const TimeRuler: React.FC<TimeRulerProps> = ({ end, height, start, tickCount, width }) => {
-  const yScale = scaleTime().domain([start, end]).range([0, height]);
```

Comment:
What would happen if we turn `start` and `end` into timezoned `Date`s before passing them? Would the ticks use the date's timezone then?

Reply:
That was one of the options I explored, using it combine with the `useKibanaTimezoneSetting`hook:
```js
const timeZone = useKibanaTimeZoneSetting();
  
const startWithOffset = new Date(new Date(start).toLocaleString('en-US', { timeZone }));
const endWithOffset = new Date(new Date(end).toLocaleString('en-US', { timeZone }));
```

This was returning correctly the expected dates. 
However, in case the set timezone is Browser, the hook returns the "local" string value, which is not an accepted value for the `.toLocaleString` method, and it would require a different approach to not pass the timeZone option under this case.

I preferred the current approach because it doesn't depend on custom values provided by this hook or the setting itself, but let me know if you were thinking of anything else to solve it using the Date object and I'll check it out :)

Reply:
Yep naming helps a lot in this case. Also, this parsing is happening at the lowest level in the tree, as the computed data is directly derived from the UTC timestamp and passed to the presentational component, we should be safe using it here.

Reply:
Thinking about it some more:

- The reliance on the global momentjs state seems brittle and needs special treatment in unit tests.
- We're handing "incorrect" timestamps to the d3 time scale, so I wonder if its tick calculation will still work in edge cases like leap years and exotic DST transitions.

The special treatment for "Browser" doesn't seem so bad if we put it into a hook such as `const startDate = useZonedDate(timestamp);`. We should be able to get around the string formatting and parsing approach by using `moment(time).tz(timeZone).toDate()`.

Does this make sense?

Reply:
It does make sense 👌 
However trying with this approach and creating the `useZonedDate` as follows:
```js
const useZonedDate = (timestamp) => {
  const timeZone = useKibanaTimeZoneSetting();

  return moment(timestamp).tz(timeZone).toDate();
};

...

 const startWithOffset = useZonedDate(start);
 const endWithOffset = useZonedDate(end);

 const yScale = scaleTime().domain([startWithOffset, endWithOffset]).range([0, height]);

```
The generated dates passed to `domain` are not parsed as time zoned and set the time ruler to the browser timezone.
Am I missing something about the approach you suggested? Anyway, I'll switch the implementation to the above comment parsing the date in case we verify that `moment(time).tz(timeZone).toDate()` is not working for this case.

Reply:
No, you're not missing anything. I mixed up my languages and assumed `Date` would maintain the time zone information in a sensible way. :see_no_evil: Sorry for the misdirection.

Reply:
No prob at all, you pointed good reason to switch anyway to Date objects, I updated the code accordingly, thanks for the review 🙏